### ISSUE TITLE
Adding a validator 

### ISSUE THREAD

**Issue by martijn10kb** [dolanmiu/docx#2395]
Currently it is possible to add a paragraph inside a paragraph. 
And having documents with more then 200 pages in a production environment makes it sometimes quite cumbersome to debug what went wrong. Especially because of word failing to give any error message whatsoever. 

It would be really nice if there would be a validator that raises an exception as soon as you try to add a paragraph inside a paragraph, or any other illegal action.

**Comment by dolanmiu** [comment]
It should be captured by type definitions, so it shouldn't be possible to add a paragraph inside a paragraph. It wouldn't compile

**Comment by martijn10kb** [comment]
No, this is not true at all. We use the type definitions heavily, but we
are still able to do this very easily

Op vr 27 okt. 2023 00:14 schreef Dolan ***@***.***>:

**Comment by dolanmiu** [comment]
Oh I see

Then there is something wrong with the type definitions.

Regarding a validator: docx uses https://github.com/ChristophWurst/xmllint-action to validate the generated xml against https://github.com/dolanmiu/docx/blob/master/ooxml-schemas/microsoft/wml-2010.xsd

Maybe you could implement something similar?
